op-challenger/game/fault/trace/cannon/provider_test.go

```diff
@@ -129,6 +129,49 @@ func TestGetStepData(t *testing.T) {
 		require.Nil(t, data)
 	})
 
+	t.Run("ReadLastStepFromDisk", func(t *testing.T) {
+		provider, initGenerator := setupWithTestData(t, dataDir, prestate)
+		initGenerator.finalState = &mipsevm.State{
+			Memory: &mipsevm.Memory{},
+			Step:   10,
+			Exited: true,
+		}
+		initGenerator.proof = &proofData{
+			ClaimValue:   common.Hash{0xaa},
+			StateData:    []byte{0xbb},
+			ProofData:    []byte{0xcc},
+			OracleKey:    common.Hash{0xdd}.Bytes(),
+			OracleValue:  []byte{0xdd},
+			OracleOffset: 10,
+		}
+		_, _, _, err := provider.GetStepData(context.Background(), 7000)
+		require.NoError(t, err)
+		require.Contains(t, initGenerator.generated, 10, "should have tried to generate the proof")
```

Comment:
This fails for me locally - it actually generates at step 7000 since it doesn't know what the final step is yet.  I'm not really sure how this isn't failing in CI...

Reply:
Oh, it fails when running just this test case individually but passes otherwise.  We shouldn't be reusing the call to `dataDir, prestate := setupTestData(t)` across all these subtests.

Reply:
It doesn't have the `test_data` in ci, nor do I have it locally which is why it was passing